src/cljs/main/broadfcui/common/components.cljs

```diff
@@ -616,3 +617,37 @@
                                  (icons/icon {} (if (:collapsed? @state) :expand :collapse)))
           body]
          body)))})
+
+(react/defc QueueStatus
+  {:render
+   (fn [{:keys [state this]}]
+     (let [{:keys [queue-status queue-error]} @state
+           {:keys [queue-time queue-position queued active]} queue-status]
+       [:div {:style {:marginBottom "0.5em"}}
+        (cond
+          queue-error (style/create-server-error-message queue-error)
+          (not queue-status) [Spinner {:text "Loading submission queue status..."}]
+          :else
+          [:div {}
+           (this :-row "Estimated wait time:" (duration/fuzzy-time-from-now-ms (+ (.now js/Date) queue-time) false))
+           (this :-row "Workflows ahead of yours:" queue-position)
+           (this :-row "Queue status:" (str queued " Queued; " active " Active")
+                 [:div {:style {:display "inline-block" :marginLeft "1ex" :fontStyle "italic"}} (links/create-internal {:data-test-id "queue-status-refresh" :onClick (fn [] (this :-load-data) (swap! state assoc :queue-status nil))} "(refresh)")])])]))
+   :component-did-mount
+   (fn [{:keys [this]}]
+     (this :-load-data))
+   :-load-data
+   (fn [{:keys [state]}]
+     (endpoints/call-ajax-orch
+      {:endpoint (endpoints/submissions-queue-status)
+       :on-done (fn [{:keys [success? status-text get-parsed-response]}]
+                  (if success?
+                    (swap! state assoc :queue-status (common/queue-status-counts (get-parsed-response false)))
```

Comment:
`common/queue-status-counts` is only called from here. Does it need to be a standalone function? Perhaps it should be integrated into this component.